### DISCUSSION TITLE
Symlinking `ark` packaged with Positron on MacOS

### DISCUSSION THREAD

**Discussion by noamross** [posit-dev/positron#4823] [Q&A] [answered]
Hi! Quick question: As ark is distributed with Positron, is the "right" way to install it locally on MacOS to symlink `/Applications/Positron.app/Contents/Resources/app/extensions/positron-r/resources/ark/ark` to a directory in the PATH (and then run `ark --install`)?  This is a pretty common pattern for MacOS apps that have some command-line component, and this way we don't have to make a security exception for it.  I just did this to get it working in Zed (where it's already the [documented R kernel](https://zed.dev/docs/repl#r-ark)!) 

If so I could add this optional step to the [Homebrew recipe](https://github.com/Homebrew/homebrew-cask/blob/1427a75d0357a19fb42b01f29089d6c9d1053e12/Casks/p/positron.rb).

**Comment by jmcphers** [comment, marked as answer]
Unless I'm misunderstanding, you don't need to symlink it to the `$PATH`; when you run `--install` it'll place the fully qualified path in the `kernel.json` file. 

I think the "right" way to install Ark would be to use one of the standalone [Ark release binaries](https://github.com/posit-dev/ark/releases/); today those are just bundled up in Positron, but it is possible that at some future date Positron's copy of Ark will diverge from the standalone version. I would expect that it will always work with other frontends, but it may also include some Positron specific bits or be refactored in other ways. You can read more about these plans here: https://github.com/posit-dev/positron/issues/3608

Thank you for maintaining the Homebrew recipe!

**Reply by noamross** [reply]
> Thank you for maintaining the Homebrew recipe!
I don't, just was going to PR ;)  (Though I am responsible for `rstudio-daily`)
